source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -12,6 +12,116 @@ namespace Extensions {
 namespace HttpFilters {
 namespace ExtAuthz {
 
+FilterConfig::FilterConfig(const envoy::config::filter::http::ext_authz::v2alpha::ExtAuthz& config,
+                           const LocalInfo::LocalInfo& local_info, const Runtime::Loader& runtime,
+                           Stats::Scope& scope, Upstream::ClusterManager& cm)
+    : allowed_request_headers_(
+          toRequestHeader(config.http_service().authorization_request().allowed_headers())),
+      allowed_client_headers_(
+          toClientHeader(config.http_service().authorization_response().allowed_client_headers())),
+      allowed_upstream_headers_(toUpstreamHeader(
+          config.http_service().authorization_response().allowed_upstream_headers())),
+      authorization_headers_to_add_(toAuthorizationHeaderToAdd(
+          config.http_service().authorization_request().headers_to_add())),
+      local_info_(local_info), runtime_(runtime),
+      cluster_name_(config.grpc_service().envoy_grpc().cluster_name()), scope_(scope), cm_(cm),
+      failure_mode_allow_(config.failure_mode_allow()) {}
+
+std::vector<Matchers::StringMatcher>
+FilterConfig::toRequestHeader(const envoy::type::matcher::ListStringMatcher& matchers) {
+  std::vector<Http::LowerCaseString> default_keys{
+      Http::Headers::get().Authorization, Http::Headers::get().Method, Http::Headers::get().Path,
+      Http::Headers::get().Host};
+  std::vector<Matchers::StringMatcher> list_matcher;
+  list_matcher.reserve(matchers.patterns().size() + default_keys.size());
+  for (const auto& key : default_keys) {
+    envoy::type::matcher::StringMatcher matcher;
+    matcher.set_exact(key.get());
+    list_matcher.push_back(matcher);
+  }
+  for (const auto& matcher : matchers.patterns()) {
```

Comment:
Couldn't you just add the defaults to the passed in matchers? Or at least copy construct a new list and then add the defaults? Is the loop necessary? Same below.